src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/import.py

```diff
@@ -36,53 +34,40 @@ def acr_import(cmd,
         cmd, registry_name, resource_group_name, IMPORT_NOT_SUPPORTED)
 
     if not source:
-        raise CLIError(INVALID_SOURCE_IMAGE)
+        raise CLIError(SOURCE_NOT_FOUND)
     source_image = source
 
-    slash = source.find('/')
-
     ImportImageParameters, ImportSource, ImportMode = cmd.get_models(
         'ImportImageParameters', 'ImportSource', 'ImportMode')
-    if slash < 0:
-        if not source_registry:
-            from knack.prompting import prompt, NoTTYException
-            try:
-                source_registry = prompt(SOURCE_REGISTRY_MISSING)
-            except NoTTYException:
-                raise CLIError(NO_TTY_ERROR)
-        if not is_valid_resource_id(source_registry):
+
+    if source_registry:
+        if is_valid_resource_id(source_registry):
+            source = ImportSource(resource_id=source_registry, source_image=source_image)
+        else:
             registry = get_registry_from_name_or_login_server(cmd.cli_ctx, source_registry, source_registry)
             if registry:
-                source_registry = registry.id
-        source = ImportSource(resource_id=source_registry, source_image=source_image)
+                source = ImportSource(resource_id=registry.id, source_image=source_image)
+            else:
+                raise CLIError(SOURCE_REGISTRY_NOT_FOUND)
     else:
-        source_registry_login_server = source_image[:slash]
-        source_image = source_image[slash + 1:]
-        if not source_image or not source_registry_login_server:
-            raise CLIError(INVALID_SOURCE_IMAGE)
-        registry = get_registry_from_name_or_login_server(cmd.cli_ctx, source_registry_login_server)
-        if registry:
-            if source_registry and \
-               source_registry.lower() != registry.id.lower() and \
-               source_registry.lower() != registry.name.lower() and \
-               source_registry.lower() != registry.login_server.lower():
-                raise CLIError(REGISTRY_MISMATCH)
-            source = ImportSource(resource_id=registry.id,
-                                  source_image=source_image)
-        else:
+        slash = source_image.find('/')
+        if slash > 0:
+            registry_uri = source_image[:slash]
```

Comment:
didn't want to check if this was possibly a host name (i.e. contains a '.')?

Reply:
what do you want to do if source registry is not provided and the string before slash does not contain a dot?

Reply:
Any valid host name could potentially work is what I had on the Team channel. /cc @sajayantony

Reply:
It should error out, a valid host name will have a dot in it.